src/peft/tuners/ia3/config.py

```diff
@@ -78,3 +80,10 @@ def __post_init__(self):
         self.feedforward_modules = (
             set(self.feedforward_modules) if isinstance(self.feedforward_modules, list) else self.feedforward_modules
         )
+
+        # check if feedforward_modules is a subset of target_modules
+        if self.feedforward_modules and self.target_modules:
+            # run a check only if feedforward_modules and target_modules are lists, and not regex expressions
+            if (not isinstance(self.feedforward_modules, str)) and (not isinstance(self.target_modules, str)):
```

Comment:
If we change the check to:

```python
if isinstance(self.feedforward_modules, set) and isinstance(self.target_modules, set):
```

It should have the same effect, right? Then, we can also get rid of the first check `if self.feedforward_modules and self.target_modules`, making the whole code a bit easier to understand. WDYT?

Reply:
Yes this was very awkward. Have cleaned this up.